### PR TITLE
Remove CLSCompliant attribute

### PR DESCRIPTION
**Pull request by JeremyKuhne**
It doesn't really matter anymore and is creating extra maintenance cost as we add new APIs. Remove versioning.targets as it did very little and all over the override attributes.

See a discussion on this here: https://github.com/dotnet/runtime/issues/44194

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12469)

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -3,9 +3,12 @@
   <Import Project="$(RepositoryEngineeringDir)Analyzer.targets" />
   <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
   <Import Project="$(RepositoryEngineeringDir)packageContent.targets" />
-  <Import Project="$(RepositoryEngineeringDir)versioning.targets" />
   <Import Project="$(RepositoryEngineeringDir)Test.targets" Condition="'$(IsTestProject)' == 'true' "/>
 
+  <PropertyGroup>
+    <Description Condition="'$(Description)' == ''">$(AssemblyName)</Description>
```
Comment (Tanya-Solyanik) [inline Directory.Build.targets:9]: What is this used for

File: src/System.Drawing.Common/src/CompatibilitySuppressions.xml [hunk 2]
```diff
@@ -25,13 +25,34 @@
...
+    <IsBaselineSuppression>true</IsBaselineSuppression>
+  </Suppression>
   <Suppression>
     <DiagnosticId>CP0014</DiagnosticId>
     <Target>M:System.Drawing.Printing.PrintPageEventArgs.get_Graphics:[T:System.Runtime.CompilerServices.NullableContextAttribute]</Target>
     <Left>lib/net8.0/System.Drawing.Common.dll</Left>
     <Right>lib/net8.0/System.Drawing.Common.dll</Right>
     <IsBaselineSuppression>true</IsBaselineSuppression>
   </Suppression>
+  <Suppression>
```
Comment (Tanya-Solyanik) [inline src/System.Drawing.Common/src/CompatibilitySuppressions.xml:42]: What does this suppression do

File: eng/versioning.targets [hunk 3]
```diff
@@ -1,35 +0,0 @@
...
-
-  <PropertyGroup Condition="'$(GenerateAssemblyInfo)'=='true'">
-    <AssemblyInfoFile Condition="'$(MSBuildProjectExtension)' == '.csproj'">$(IntermediateOutputPath)_AssemblyInfo.g.cs</AssemblyInfoFile>
-  </PropertyGroup>
-
-  <PropertyGroup>        
-    <Description Condition="'$(Description)' == ''">$(AssemblyName)</Description>
-  </PropertyGroup>
-
-  <!-- Note: ReferenceAssemblies.targets still uses @(AssemblyInfoLines) as well. So if you remove this, those need to migrate too. -->
```
Comment (Tanya-Solyanik) [inline eng/versioning.targets:14]: Had you validated that referenceassemblies.targets should not be updated?

**Review by Tanya-Solyanik — APPROVED** [review]

File: src/System.Drawing.Common/src/CompatibilitySuppressions.xml (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Drawing.Common/src/CompatibilitySuppressions.xml:42]: API Compat tool

File: eng/versioning.targets (hunk 3 above)
Comment (JeremyKuhne) [reply eng/versioning.targets:14]: Yes

File: src/System.Windows.Forms/src/System/Windows/Forms/OLE/DataObject.DataStore.cs [hunk 4]
```diff
@@ -9,7 +9,7 @@ namespace System.Windows.Forms;
 
 public partial class DataObject
 {
-    private class DataStore : IDataObject
+    private sealed class DataStore : IDataObject
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/OLE/DataObject.DataStore.cs:12]: Whoops- going to leave it in so I don't invalidate the approval.

**Pull request merged**